### PR TITLE
#27776: nailed the client modules together and adapted them to Python 3

### PR DESCRIPTION
**Pull request by magenta-af**
Just in time for [the upcoming deprecation of Python 2](https://pythonclock.org/), this branch adapts the client modules to Python 3, merges them into one package, tidies them up very slightly, and removes as many uses of the word `bibos` as I could manage without breaking things. There's also a fix for #30644, the annoying PyYAML warning, in here too.

This works fine on (one of) our test machines, and I have a separate script (destined, presumably, for [OS2borgerPC/bibos_image](https://github.com/OS2borgerPC/bibos_image)) that tries to upgrade to this new package and reinstalls the old ones in a panic if something goes wrong.

(Of course, we still need to port all of the _scripts_ to Python 3, but that's a task for another day...)

### CODE REVIEW

File: os2borgerpc_client/bin/register_new_os2borgerpc_client.sh [hunk 1]
```diff
@@ -112,47 +129,53 @@
...
+        fatal "Tilmelding mislykkedes" && continue || exit 1
+    fi
 
     # Now setup cron job
     if [[ -f $(which jobmanager) ]]
     then
-        sudo echo 'PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' > /etc/cron.d/bibos-jobmanager
-        sudo echo "*/5 * * * * root $(which jobmanager)" >> /etc/cron.d/bibos-jobmanager
+        echo 'PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' > /etc/cron.d/os2borgerpc-jobmanager
+        echo "*/5 * * * * root $(which jobmanager)" >> /etc/cron.d/os2borgerpc-jobmanager
```
Comment (dals83) [inline os2borgerpc_client/bin/register_new_os2borgerpc_client.sh]: We have a script that randomizes the 5 minute interval: https://bibos-admin.magenta-aps.dk/site/hvidovre/scripts/355/

Could we reuse it here?

File: os2borgerpc_client/setup.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from setuptools import setup
+
+setup(
+    name='os2borgerpc_client',
+    # Keep this in sync with os2borgerpc_client/jobmanager.py
+    version='0.0.5.1',
```
Comment (dals83) [inline os2borgerpc_client/setup.py]: I think we should bump the version to 0.1.0.0

File: os2borgerpc_client/setup.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        'bin/os2borgerpc_find_gateway',
+        'bin/os2borgerpc_register_in_admin',
+        'bin/os2borgerpc_push_config_keys',
+        'bin/os2borgerpc_upload_dist_packages',
+        'bin/os2borgerpc_upload_packages',
+        'bin/jobmanager',
+        'bin/register_new_os2borgerpc_client.sh',
+        'bin/admin_connect.sh',
+
+        # Compatiblity symlinks
```
Comment (dals83) [inline os2borgerpc_client/setup.py]: Compatiblity = Compatibility :)

File: os2borgerpc_client/bin/os2borgerpc_register_in_admin [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
-config = BibOSConfig()
+config = OS2borgerPCConfig()
 data = config.get_data()
 
 mac = data['mac']
 hostname = data['hostname']
 site = data['site']
 distribution = data['distribution']
-admin_url = data.get('admin_url', 'http://bibos.magenta-aps.dk/')
+admin_url = data.get('admin_url', 'http://os2borgerpc.magenta-aps.dk/')
```
Comment (dals83) [inline os2borgerpc_client/bin/os2borgerpc_register_in_admin]: Okay we need to add this as an alias to our production server...

Comment (magenta-af) [reply os2borgerpc_client/bin/os2borgerpc_register_in_admin]: Yeah, this one is weird. We _already_ don't have a system called `bibos.magenta-aps.dk`...

(`register_new_os2borgerpc_client.sh` will actually always override this default value, so we never use it. Should we do something about that?)

File: os2borgerpc_client/os2borgerpc/client/config.py [hunk 5]
```diff
@@ -4,8 +4,7 @@
 import sys
 import stat
 
-DEFAULT_CONFIG_FILES = [
-        "/etc/os2borgerpc/os2borgerpc.conf", "/etc/bibos/bibos.conf"]
+DEFAULT_CONFIG_FILES = ["/etc/os2borgerpc/os2borgerpc.conf"]
```
Comment (magenta-af) [inline os2borgerpc_client/os2borgerpc/client/config.py:7]: I approve of removing the BibOS branding, but isn't this particular change dangerous? What will happen if we upgrade a system with `/etc/bibos/bibos.conf` to a version of the client package that no longer reads that file?

File: os2borgerpc_client/setup.py (hunk 2 above)
Comment (magenta-af) [reply os2borgerpc_client/setup.py:6]: @agger-magenta _Should_ we bump the version number a bit more before closing this?

File: os2borgerpc_client/bin/register_new_os2borgerpc_client.sh (hunk 1 above)
Comment (magenta-af) [reply os2borgerpc_client/bin/register_new_os2borgerpc_client.sh:179]: I suppose we could, but I think I'm fine keeping the existing behaviour for now...

@agger-magenta, what do you think?

File: os2borgerpc_client/bin/os2borgerpc_register_in_admin (hunk 4 above)
Comment (magenta-af) [reply os2borgerpc_client/bin/os2borgerpc_register_in_admin:17]: What _is_ this server called now? (And shouldn't the URL be `https` in all cases?)

**Review by magenta-af — COMMENTED** [review]
I'm technically the author of this PR, so I can't approve it. :-)

I have no other comments, though, so: approved (with some minor qualifications).

File: os2borgerpc_client/os2borgerpc/client/config.py (hunk 5 above)
Comment (agger-magenta) [reply os2borgerpc_client/os2borgerpc/client/config.py:7]: The we must move it.

File: os2borgerpc_client/setup.py (hunk 2 above)
Comment (agger-magenta) [reply os2borgerpc_client/setup.py:6]: We will do that when we release, i.e. when we merge to master and tag it.

File: os2borgerpc_client/bin/register_new_os2borgerpc_client.sh (hunk 1 above)
Comment (agger-magenta) [reply os2borgerpc_client/bin/register_new_os2borgerpc_client.sh:179]: I agree with Alec.

File: os2borgerpc_client/bin/os2borgerpc_register_in_admin (hunk 4 above)
Comment (agger-magenta) [reply os2borgerpc_client/bin/os2borgerpc_register_in_admin:17]: Yeah, this should probably be fixed up a bit.

File: os2borgerpc_client/os2borgerpc/client/config.py (hunk 5 above)
Comment (agger-magenta) [reply os2borgerpc_client/os2borgerpc/client/config.py:7]: No, you're actually right. I'll keep the `bibos` location for backwards compatibility.

File: os2borgerpc_client/setup.py (hunk 2 above)
Comment (agger-magenta) [reply os2borgerpc_client/setup.py:6]: I think we're changing the numbering to ordinary semantic versioning and bumping to 1.0.0 for starters.

It's kind of mature enough already. :slightly_smiling_face:

**Review by agger-magenta — APPROVED** [review]

**Pull request merged**